pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -517,15 +517,34 @@ func validateAzure(h *validatorHandler, m *Machine) (bool, utilerrors.Aggregate)
 	}
 
 	if providerSpec.Image.ResourceID == "" {
-		errs = append(errs, field.Required(field.NewPath("providerSpec", "image", "resourceID"), "resourceID must be provided"))
+		imageProvided := true
```

Comment:
I'm not following this, why `imageProvided := true` when `providerSpec.Image.ResourceID == ""`?

Reply:
Now this won't error when providerSpec.Image.ResourceID == ""? Why?

Reply:
Let me rewrite this, it actually does not seem clear. But this is due to logic in 

https://github.com/openshift/cluster-api-provider-azure/blob/master/pkg/cloud/azure/services/virtualmachines/virtualmachines.go#L105-L115

Reply:
I'm going to adjust the support in https://github.com/openshift/cluster-api-provider-azure/pull/140, so the options will be equivalent, and you could either specify `imageReference` by offer etc. or give it a recourceID.

Reply:
Could we may be make this logic less clever and more explicit so is easier to read?
And present the output as one atomic error similar to what we do here -> https://github.com/openshift/machine-api-operator/blob/master/pkg/apis/machine/v1beta1/machine_webhook.go#L331-L339